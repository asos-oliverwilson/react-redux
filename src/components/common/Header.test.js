/* eslint-disable */
import React, { Component } from 'react';
import Enzyme, { shallow, render, mount } from 'enzyme';
import Header from './Header';
import LoadingDots from './LoadingDots';

const createHeader = (isLoading) => {
    const component = shallow(<Header loading={isLoading} />);
    const LoadingDotsComponent = <LoadingDots interval={100} dots={20} />;
    return {
        wrapper: component,
        LoadingDotsComponent: LoadingDotsComponent
    };
};

describe("COMPONENT: Header", () => {
    it("should display dots if loading is true", () => {
        const {wrapper, LoadingDotsComponent} = createHeader(true);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(LoadingDotsComponent)).toEqual(true);
    });
    it("should not display dots if loading is false", () => {
        const {wrapper, LoadingDotsComponent} = createHeader(false);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(LoadingDotsComponent)).toEqual(false);
    });
});