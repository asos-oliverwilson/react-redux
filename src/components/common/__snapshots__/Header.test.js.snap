// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT: Header should display dots if loading is true 1`] = `
ShallowWrapper {
  "length": 1,
  "node": <nav>
    <IndexLink
      activeClassName="active"
      to="/"
    >
      Home
    </IndexLink>
     | 
    <Link
      activeClassName="active"
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/about"
    >
      About
    </Link>
     | 
    <Link
      activeClassName="active"
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/courses"
    >
      Courses
    </Link>
    <LoadingDots
      dots={20}
      interval={100}
    />
  </nav>,
  "nodes": Array [
    <nav>
      <IndexLink
        activeClassName="active"
        to="/"
      >
        Home
      </IndexLink>
       | 
      <Link
        activeClassName="active"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/about"
      >
        About
      </Link>
       | 
      <Link
        activeClassName="active"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/courses"
      >
        Courses
      </Link>
      <LoadingDots
        dots={20}
        interval={100}
      />
    </nav>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Header
        loading={true}
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "loading": true,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <nav>
          <IndexLink
            activeClassName="active"
            to="/"
          >
            Home
          </IndexLink>
           | 
          <Link
            activeClassName="active"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/about"
          >
            About
          </Link>
           | 
          <Link
            activeClassName="active"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/courses"
          >
            Courses
          </Link>
          <LoadingDots
            dots={20}
            interval={100}
          />
        </nav>,
        "_debugID": 2,
        "_renderedOutput": <nav>
          <IndexLink
            activeClassName="active"
            to="/"
          >
            Home
          </IndexLink>
           | 
          <Link
            activeClassName="active"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/about"
          >
            About
          </Link>
           | 
          <Link
            activeClassName="active"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/courses"
          >
            Courses
          </Link>
          <LoadingDots
            dots={20}
            interval={100}
          />
        </nav>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Header
    loading={true}
  />,
}
`;

exports[`COMPONENT: Header should not display dots if loading is false 1`] = `
ShallowWrapper {
  "length": 1,
  "node": <nav>
    <IndexLink
      activeClassName="active"
      to="/"
    >
      Home
    </IndexLink>
     | 
    <Link
      activeClassName="active"
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/about"
    >
      About
    </Link>
     | 
    <Link
      activeClassName="active"
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/courses"
    >
      Courses
    </Link>
  </nav>,
  "nodes": Array [
    <nav>
      <IndexLink
        activeClassName="active"
        to="/"
      >
        Home
      </IndexLink>
       | 
      <Link
        activeClassName="active"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/about"
      >
        About
      </Link>
       | 
      <Link
        activeClassName="active"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/courses"
      >
        Courses
      </Link>
    </nav>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Header
        loading={false}
      />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "loading": false,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <nav>
          <IndexLink
            activeClassName="active"
            to="/"
          >
            Home
          </IndexLink>
           | 
          <Link
            activeClassName="active"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/about"
          >
            About
          </Link>
           | 
          <Link
            activeClassName="active"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/courses"
          >
            Courses
          </Link>
        </nav>,
        "_debugID": 4,
        "_renderedOutput": <nav>
          <IndexLink
            activeClassName="active"
            to="/"
          >
            Home
          </IndexLink>
           | 
          <Link
            activeClassName="active"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/about"
          >
            About
          </Link>
           | 
          <Link
            activeClassName="active"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/courses"
          >
            Courses
          </Link>
        </nav>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Header
    loading={false}
  />,
}
`;
